src/varlinkctl/varlinkctl.c

```diff
@@ -530,13 +541,23 @@ static int verb_call(int argc, char *argv[], void *userdata) {
         _cleanup_(sd_varlink_unrefp) sd_varlink *vl = NULL;
         const char *url, *method, *parameter, *source;
         unsigned line = 0, column = 0;
+        char **cmdline;
         int r;
 
         assert(argc >= 3);
-        assert(argc <= 4);
+
+        if (argc > 4 && !arg_exec)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Too many arguments.");
+        if (arg_exec && argc < 5)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Expected command line to execute.");
+
+        if (arg_exec && (arg_collect || (arg_method_flags & (SD_VARLINK_METHOD_ONEWAY|SD_VARLINK_METHOD_MORE))) != 0)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "--exec and --collect/--more/--oneway may not be combined.");
+
         url = argv[1];
         method = argv[2];
         parameter = argc > 3 && !streq(argv[3], "-") ? argv[3] : NULL;
```

Comment:
So, when `--exec`, the third argument cannot be ignored.
Maybe better to mention that in the man page.